mapie/metrics.py

```diff
@@ -171,3 +177,130 @@ def classification_mean_width_score(y_pred_set: ArrayLike) -> float:
     )
     mean_width = y_pred_set.sum(axis=1).mean()
     return float(mean_width)
+
+
+def expected_calibration_error(
+    y_true: ArrayLike,
+    y_scores: ArrayLike,
+    num_bins: int = 50,
+    split_strategy: Optional[str] = None,
+) -> float:
+    """
+    The expected calibration error, which is the difference between
+    the confidence scores and accuracy per bin [1].
+
+    [1] Naeini, Mahdi Pakdaman, Gregory Cooper, and Milos Hauskrecht.
+    "Obtaining well calibrated probabilities using bayesian binning."
+    Twenty-Ninth AAAI Conference on Artificial Intelligence. 2015.
+
+    Parameters
+    ----------
+    y_true : ArrayLike of shape (n_samples,)
+        The "true" values, target for the calibrator.
+    y_score : ArrayLike of shape (n_samples,) or (n_samples, n_classes)
+        The predictions scores.
+    num_bins : int
+        Number of bins to make the split in the y_score.
+    strategy : str
+        The way of splitting the predictions into different bins.
+    Returns
+    -------
+    float
+        The score of ECE (Expected Calibration Error).
+    """
+    split_strategy = check_split_strategy(split_strategy)
+    check_number_bins(num_bins)
+    y_true_ = check_binary_zero_one(y_true)
+    y_scores = cast(NDArray, y_scores)
+
+    if np.size(y_scores.shape) == 2:
+        y_score = cast(
+            NDArray, column_or_1d(np.max(y_scores, axis=1))
+        )
+    else:
+        y_score = cast(NDArray, column_or_1d(y_scores))
+
+    _, bin_accs, bin_confs, bin_sizes = calc_bins(
+        y_true_, y_score, num_bins, split_strategy
+    )
+
+    return np.divide(
+        np.sum(bin_sizes * np.abs(bin_accs - bin_confs)),
+        np.sum(bin_sizes)
+    )
+
+
+def top_label_ece(
+    y_true: ArrayLike,
+    y_scores: ArrayLike,
+    y_score_arg: Optional[ArrayLike] = None,
+    num_bins: int = 50,
+    split_strategy: Optional[str] = None,
+    classes: Optional[ArrayLike] = None,
+) -> float:
+    """
+    The Top-Label ECE which is a method adapted to fit the
+    ECE to a Top-Label setting. [2].
+
+    [2] Gupta, Chirag, and Aaditya K. Ramdas.
+    "Top-label calibration and multiclass-to-binary reductions."
+    arXiv preprint arXiv:2107.08353 (2021).
+
+    Parameters
+    ----------
+    y_true : ArrayLike of shape (n_samples,)
+        The "true" values, target for the calibrator.
+    y_scores : ArrayLike of shape (n_samples, n_classes)
+    or (n_samples,)
+        The predictions scores, either the maximum score and the
+        argmax needs to be inputted or in the form of the prediction
+        probabilities.
+    y_score_arg : ArrayLike of shape (n_samples,)
+        If only the maximum is provided in the y_scores, the argmax must
+        be provided here.
+    num_bins : int
+        Number of bins to make the split in the y_score.
+    strategy : str
+        The way of splitting the predictions into different bins.
```

Comment:
same remark,specify allowed valued in the docstring

Reply:
done.